torch/fx/passes/runtime_assert.py

```diff
@@ -34,6 +44,161 @@ def _get_example_value(node: fx.Node) -> Optional[str]:
         return None
 
 
+def _get_sym_val(node: fx.Node) -> "Optional[sympy.Expr]":
+    """
+    Returns sympy.Expr from node example value
+    """
+    if (val := _get_example_value(node)) is not None:
+        # dynamo has py_sym_type, export uses sympy.Expr
+        if isinstance(val, py_sym_types):
+            return val.node.expr
+        elif isinstance(val, sympy.Expr):
+            return val
+    return None
+
+
+def _get_sym_rel(expr):
+    """
+    Enumerates supported relations
+    """
+    if isinstance(expr, sympy.LessThan):
+        return SymRel.LE
+    if isinstance(expr, sympy.StrictLessThan):
+        return SymRel.LT
+    elif isinstance(expr, sympy.GreaterThan):
+        return SymRel.GE
+    elif isinstance(expr, sympy.StrictGreaterThan):
+        return SymRel.GT
+    elif isinstance(expr, sympy.Equality):
+        return SymRel.EQ
+    elif isinstance(expr, sympy.Unequality):
+        return SymRel.NE
+    return None
+
+
+def _sym_rel_to_sympy(rel, lhs, rhs):
+    """
+    Sort of inverse of _get_sym_rel()
+    """
+    if rel == SymRel.EQ:
+        return sympy.Equality(lhs, rhs)
+    elif rel == SymRel.NE:
+        return sympy.Unequality(lhs, rhs)
+    if rel == SymRel.LE:
+        return sympy.LessThan(lhs, rhs)
+    elif rel == SymRel.LT:
+        return sympy.StrictLessThan(lhs, rhs)
+    elif rel == SymRel.GE:
+        return sympy.GreaterThan(lhs, rhs)
+    elif rel == SymRel.GT:
+        return sympy.StrictGreaterThan(lhs, rhs)
+
+
+def _canonicalize_expr(expr, mapping):
+    """
+    https://github.com/pytorch/pytorch/pull/128411 means we won't get expressions like -u0 <= 0
+    for calls like torch._check(-u0 <= 0) anymore, but it does seem like we can get 4 <= u0, or u0 < s0,
+    which are slightly difficult to handle when we want to track ranges.
+    Here we move all non-numeric terms to the LHS, and maybe negate the expression if is_size = False,
+    in case we already track the negated value.
+    """
+    if expr is None or len(expr.args) != 2:
+        return expr
+    # assume we won't get something like u0 + 2 < 4, so don't call this
+    # expr = sympy.simplify(expr)
+
+    # if straightforward we return
+    if isinstance(expr.args[1], sympy.Number):
+        return expr
+
+    # try flipping, won't hurt
+    # can return if original expr is something like 4 <= s0
+    lhs, rhs = expr.args
+    if isinstance(expr, sympy.LessThan):
+        expr = sympy.GreaterThan(rhs, lhs)
+    elif isinstance(expr, sympy.StrictLessThan):
+        expr = sympy.StrictGreaterThan(rhs, lhs)
+    elif isinstance(expr, sympy.GreaterThan):
+        expr = sympy.LessThan(rhs, lhs)
```

Comment:
This operation is not idempotent?? Why do you always flip the expresion from `a > b` into `b < a` AND from `b < a` into `a > b`??

Reply:
This is not very canonical...
You don't need to reinvent the wheel here. If you want to find the additive constants in an expression, you should call canonicalize and find it filtering the args on the left / right of the expression.
Canonicalisation assures that the expression is just of the form `<` or `<=`, and that the additive constant is positive. You can then use this however you want in your bounds.